code/modules/reagents/chemistry/equilibrium.dm

```diff
@@ -0,0 +1,287 @@
+		to_delete = TRUE
+		return
+	if(!length(reaction.results)) //Come back to and revise the affected reactions in the next PR, this is a placeholder fix.
+		holder.instant_react(reaction) //Even if this check fails, there's a backup - look inside of calculate_yield()
+		to_delete = TRUE
+		return
+	reaction.on_reaction(holder, multiplier)
+	SSblackbox.record_feedback("tally", "chemical_reaction", 1, "[reaction.type] attempts")
+	
+
+/datum/equilibrium/Destroy()
+	LAZYREMOVE(holder.reaction_list, src)
+	holder = null
+	reaction = null
+	return ..()
+
+/* 
+* Check to make sure our input vars are sensible - truncated version of check_reagent_properties() 
+* 
+* (as the setup in holder.dm checks for that already - this is a way to reduce calculations on New())
+* Don't call this unless you know what you're doing, this is an internal proc
+*/
+/datum/equilibrium/proc/check_inital_conditions()
+	//Make sure we have the right multipler for on_reaction()
+	for(var/B in reaction.required_reagents)
+		multiplier = min(multiplier, round((holder.get_reagent_amount(B) / reaction.required_reagents[B]), CHEMICAL_QUANTISATION_LEVEL))
+	if(multiplier == INFINITY) 
+		return FALSE
+	//Consider purity gating too? - probably not, purity is hard to determine
+	//To prevent reactions outside of the pH window from starting.
+	if(! ((holder.pH >= (reaction.optimal_pH_min - reaction.determin_pH_range)) && (holder.pH <= (reaction.optimal_pH_max + reaction.determin_pH_range)) ))
+		return FALSE
+	return TRUE
+
+/* 
+* Check to make sure our input vars are sensible - is the holder overheated? does it have the required reagents? Does it have the required calalysts?
+*
+* If you're adding more checks for reactions, this is the proc to edit
+* otherwise, generally, don't call this directed except internally
+*/
+/datum/equilibrium/proc/check_reagent_properties()
+	//Have we exploded?
+	if(!holder.my_atom || holder.reagent_list.len == 0)
+		return FALSE
+	//Are we overheated?
+	if(holder.chem_temp > reaction.overheat_temp) //This is before the process - this is here so that overly_impure and overheated() share the same code location (and therefore vars) for calls.
+		SSblackbox.record_feedback("tally", "chemical_reaction", 1, "[reaction.type] overheated reaction steps")
+		reaction.overheated(holder, src)
+
+	//set up catalyst checks
+	var/total_matching_catalysts = 0
+	//Reagents check should be handled in the calculate_yield() from multiplier
+
+	//If the product/reactants are too impure
+	for(var/r in holder.reagent_list)
+		var/datum/reagent/R = r
+		if (R.purity < reaction.purity_min)//If purity is below the min, call the proc
+			SSblackbox.record_feedback("tally", "chemical_reaction", 1, "[reaction.type] overly impure reaction steps")
+			reaction.overly_impure(holder, src)
+		//this is done this way to reduce processing compared to holder.has_reagent(P)
+		for(var/P in reaction.required_catalysts)
+			var/datum/reagent/R0 = P
```

Comment:
Please use better variable names than `R` and `R0`. Whenever you see yourself in need of using `R0` is the moment you know you should be using something more clear. `catalyst` works here

Reply:
Oops, I forgot I did that, changed.